nat-lab/tests/test_notification_center.py

```diff
@@ -35,23 +42,56 @@ def verify_mqtt_payload(payload):
         verify_uuid(machine_uuid)
 
 
-def run_mqtt_listener():
-    payload_queue: queue.Queue[str] = queue.Queue(1)
-
-    def on_message(_client, _userdata, message):
-        payload_queue.put(f"{message.payload.decode()}")
-
-    mqttc = mqtt.Client(client_id="reciever", protocol=mqtt.MQTTv311)
-
-    mqttc.on_message = on_message
-
-    mqttc.tls_set(certfile=None, keyfile=None, cert_reqs=ssl.CERT_NONE)
-    mqttc.connect(MQTT_BROKER_IP, port=8883, keepalive=1)
-    mqttc.subscribe("meshnet", qos=0)
-
-    mqttc.loop_start()
-
-    return payload_queue, mqttc
+async def run_mqtt_listener(
+    exit_stack,
+    connection,
+    mqtt_broker_host,
+    mqtt_broker_port,
+    mqtt_broker_user,
+    mqtt_broker_password,
+):
+    mqtt_process = await exit_stack.enter_async_context(
+        connection.create_process([
+            "python3",
+            "/opt/bin/mqtt-listener.py",
+            mqtt_broker_host,
+            mqtt_broker_port,
+            mqtt_broker_user,
+            mqtt_broker_password,
+        ]).run()
+    )
+    await sleep(1)
+    return mqtt_process
+
+
+async def get_mqtt_broker_credentials(connection):
+    endpoint = f"{CORE_API_URL}/v1/notifications/token"
+    json_data = {
+        "user_hash": "user-hash",
+        "app_user_uid": "app-user-uid",
+        "platform_id": 1,
+        "protocol": "tcp",
+    }
+    payload = json.dumps(json_data)
+    response = await send_https_request(
+        connection,
+        endpoint,
+        "POST",
+        CORE_API_CA_CERTIFICATE_PATH,
+        payload,
+        authorization_header=basic_authorization_header,
+    )
+    assert "endpoint" in response
+    assert "username" in response
+    assert "password" in response
+    assert "expires_in" in response
+
+    mqtt_data = response["endpoint"].replace("//", "").split(":")
+    host = mqtt_data[1]
+    port = mqtt_data[2]
```

Comment:
Nit: tbh you could just do `_, host, port = ` (what is in 0, though?)

Reply:
0 - is protocol